cli/src/pcluster/templates/login_nodes_stack.py

```diff
@@ -0,0 +1,217 @@
+from typing import Dict
+
+from aws_cdk import aws_autoscaling as autoscaling
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_elasticloadbalancingv2 as elbv2
+from aws_cdk.core import CfnTag, NestedStack, Stack
+from constructs import Construct
+
+from pcluster.config.cluster_config import LoginNodesPools, SlurmClusterConfig
+from pcluster.templates.cdk_builder_utils import (
+    CdkLaunchTemplateBuilder,
+    LoginNodesIamResources,
+    get_default_instance_tags,
+    get_default_volume_tags,
+    get_login_nodes_security_groups_full,
+)
+from pcluster.utils import get_http_tokens_setting
+
+
+class LoginNodesStack(NestedStack):
+    """Stack encapsulating a set of LoginNodes and the associated resources."""
```

Comment:
General comment on the approach used here to manage the pool info, can we use a [Construct](https://docs.aws.amazon.com/cdk/v2/guide/constructs.html) to organise this code for better readbaility?
```
class Pool(Construct):
    ...
    <pool related resources>
```

Then keep the LoginNodesStack responsible for creating a Pool construct for each pool.

```
class LoginNodesStack(NestedStack):
    ...
    for pool in pools:
        Pool(...)
    ...
```

Ref: [Writing your own constructs](https://docs.aws.amazon.com/cdk/v2/guide/constructs.html#constructs_author)

Reply:
Okay, I will try.

Reply:
Done!